render_silhouette_regmarks.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright (C) 2021 miLORD1337
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA.
+#
+"""
+Base module for rendering regmarks for Silhouette CAMEO products in Inkscape.
+"""
+
+import inkex
+from inkex import EffectExtension, Boolean, Rectangle, PathElement, Layer, Group, TextElement, Transform
+from gettext import gettext
+
+REGMARK_LAYERNAME = 'Regmarks'
+REGMARK_LAYER_ID = 'regmark'
+REGMARK_TOP_LEFT_ID = 'regmark-tl'
+REGMARK_TOP_RIGHT_ID = 'regmark-tr'
+REGMARK_BOTTOM_LEFT_ID = 'regmark-bl'
+REGMARK_SAFE_AREA_ID = 'regmark-safe-area'
+REGMARK_NOTES_ID = 'regmark-notes'
+
+REG_SQUARE_MM = 5
+REG_LINE_MM = 20
+REG_SAFE_AREA_MM = 20
+
+# https://www.reddit.com/r/silhouettecutters/comments/wcdnzy/the_key_to_print_and_cut_success_an_extensive/
+# > The registration mark thickness is actually very important. For some reason, 0.3 mm marks work perfectly. 
+# > The thicker you get, the less accurate registration will be. ~~~ galaxyman47
+REG_MARK_LINE_WIDTH_MM = 0.3
+
+REG_MARK_INFO_FONT_SIZE_PX = 2.5
+
+ENABLE_CHECKERBOARD = True
+
+class InsertRegmark(EffectExtension):
+	def add_arguments(self, pars):
+		# Parse arguments
+		pars.add_argument("-X", "--reg-x", "--regwidth",  type = float, dest = "regwidth",   default = 180.0, help="X mark distance [mm]")
+		pars.add_argument("-Y", "--reg-y", "--reglength", type = float, dest = "reglength",  default = 230.0, help="Y mark distance [mm]")
+		pars.add_argument("--rego-x",  "--regoriginx",    type = float, dest = "regoriginx", default = 15.0,  help="X mark origin from left [mm]")
+		pars.add_argument("--rego-y", "--regoriginy",     type = float, dest = "regoriginy", default = 20.0,  help="X mark origin from top [mm]")
```

Comment:
wanna sync default values with INX? (0,0,10,10)
or is there a good reason to differ from it?

Reply:
Yup we should sync. Fixed